### PR TITLE
Connected admin endpoints

### PR DESCRIPTION
**Pull request by RealGorandos**
Connected create/delete user and create group endpoints....(add/remove user to a group needs to change the passed body in the frontend)

### CODE REVIEW

**Review by Abdulla-Alkhulaqui — APPROVED** [review]

**Comment by Abdulla-Alkhulaqui** [comment]
@Mohido  there are some changes in the backend in this PR, please review them.

File: backend/routes/sections.js [hunk 1]
```diff
@@ -65,14 +65,13 @@ router.delete(
...
     let unsuccess = [];
-    let result1 = await deleteFromTable("grades", req.body);
-    if (result1.error) unsuccess = [...unsuccess, section];
-
-    let result2 = await deleteFromTable("tasks", req.body);
-    if (result2.error) unsuccess = [...unsuccess, section];
-
-    let result3 = await deleteFromTable("sections", req.body);
-    if (result3.error) unsuccess = [...unsuccess, section];
+    for (const section of req.body) {
```
Comment (Mohido) [inline backend/routes/sections.js:68]: Not approved

File: backend/routes/tasks.js [hunk 2]
```diff
@@ -28,9 +28,11 @@ const exec = util.promisify(require('child_process').exec);
     /* Construction of the query parameters */
     let parameters = {};
     if(req.query.sectionid)
-        parameters.sectionid = req.query.sectionid;
+        parameters.sectionID = req.query.sectionid;
+    if(req.query.groupid)
```
Comment (Mohido) [inline backend/routes/tasks.js:32]: We don't have groupids in the get tasks.

File: backend/routes/tasks.js [hunk 3]
```diff
@@ -40,12 +42,11 @@ const exec = util.promisify(require('child_process').exec);
...
         let finalShape = {
-            taskid: task.taskid,
+            taskid: task.taskID,
             taskname: task.taskname,
-            sectionid: task.sectionid,
-            max : task.max,
-            dueDate : task.expirydate,
-            dueTime : task.expirytime
+            sectionid: task.sectionID,
+            groupid : task.groupID,
```
Comment (Mohido) [inline backend/routes/tasks.js:48]: Missing data and groupid is not a column in the tables

File: backend/routes/tasks.js [hunk 4]
```diff
@@ -72,59 +73,48 @@ const exec = util.promisify(require('child_process').exec);
...
-        expiryDate : req.body.dueDate  || `${ts.getFullYear()}-${ts.getMonth()}-${ts.getDay()}`,
-        expiryTime : req.body.dueTime ||  `${ts.getHours()}:${ts.getMinutes()}:${ts.getSeconds()}`,
-        solution : solution.replace(/^.*module.*$/g,'').replace(/\'/g, "''"),
-        description: description.replace(/\'/g, "''"),
-    };
+    const solExists = (req.files && req.files.solution) || (req.body  && req.body.solution);
+    const descExists = (req.files && req.files.description) || (req.body  && req.body.description);
+    const incompleteFile = !solExists && !descExists;
+    const incompleteBody = !req.body || !req.body.taskid || !req.body.sectionid || !req.body.groupid;
+    // if(incompleteFile || incompleteBody)
```
Comment (Mohido) [inline backend/routes/tasks.js:80]: This is not the latest tasks endpoint. Please rebase before pushing

File: backend/routes/users.js [hunk 5]
```diff
@@ -132,12 +132,12 @@ router.post('/create', isAuth, protector(["admin"]), async (req, res, next) =>{
...
     /* Create users in the database. */
     for (const user of req.body.users) {
-        console.log(user);
         const student = {
             neptun: user.neptun || "",
             username: user.username, 
             firstname : user.firstname || "", 
-            lastname : user.lastname || ""
+            lastname : user.lastname || "",
+            neptun: user.username
```
Comment (Mohido) [inline backend/routes/users.js:140]: Redundent

**Review by Mohido — CHANGES_REQUESTED** [review]
Please, return the backend to the latest push. This version is old and edited. The new editions are not compatible.

**Comment by RealGorandos** [comment]
@Abdulla-Alkhulaqui  Please check the frontend after the rebase.
@Mohido  @the4t4  I have rebased the BE of this branch to development please check those files!

**Review by Mohido — APPROVED** [review]
Backend looks fine.

**Review by the4t4 — APPROVED** [review]

**Comment by Abdulla-Alkhulaqui** [comment]
> @RealGorandos Merge?

I will after merging mine.

**Comment by the4t4** [comment]
Pls squash merge merge lol, 91 commits and counting

**Comment by RealGorandos** [comment]
@Abdulla-Alkhulaqui How about merging those two then urs :)

**Pull request merged**